### ISSUE TITLE
[BUG] Password server don't work

### ISSUE THREAD

**Issue by Fr-Dae** [unitystation/unitystation#10528]
### There is no other issue reporting the same problem.

- [X] I have searched the existing issues

### Game Version

24.07.27.17

### I was

Server

### OS

Windows 8, 10, 11

### What happened?

Hello, I'm currently testing the twitch event server. (docker, on dedicated host)

My current problem is that when you set a password, it doesn't change anything, the connection remains free.

whether it's
- with the config.json file at the root of the server `“ConnectionPassword”: “password”,`”.

or 
- via the ingame admin interface in server parameters “password” and validating with enter
(there's also an admin log to say it's been taken into account)

unfortunately I can't remember who set this up, I think it was gilles, but I can't find any conversation about it. 

I asked 3 people, including @waffielz to connect to this test server, and they were able to do so without any problems. 

### What should have happened instead?

when i set a password, a windows ask it for player at join 

### Steps to reproduce the issue

start hub, clic on "join"
don't see any password windows

**Comment by MaxIsJoe** [comment]
@corp-0 I think we might have disabled server passwords or removed it while we were migrating to central command

**Comment by corp-0** [comment]
No, the password thing should be unrelated to account authentication. I will take a look nonetheless

**Comment by craftxbox** [comment]
Reproducible as of 25032122
